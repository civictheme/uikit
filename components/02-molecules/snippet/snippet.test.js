const template = 'components/02-molecules/snippet/snippet.twig';

describe('Snippet Component', () => {
  test('renders with required attributes', async () => {
    const c = await dom(template, {
      title: 'Snippet Title',
      summary: 'This is the summary of the snippet.',
    });

    expect(c.querySelectorAll('.ct-snippet')).toHaveLength(1);
    expect(c.querySelector('.ct-snippet__title').textContent.trim()).toEqual('Snippet Title');
    expect(c.querySelector('.ct-snippet__summary').textContent.trim()).toEqual('This is the summary of the snippet.');

    assertUniqueCssClasses(c);
  });

  test('renders with optional attributes', async () => {
    const c = await dom(template, {
      content_top: 'Top content',
      title: 'Snippet Title',
      summary: 'This is the summary of the snippet.',
      content_middle: 'Middle content',
      content_bottom: 'Bottom content',
      link: {
        text: 'Read more',
        url: 'https://example.com/read-more',
        is_new_window: true,
        is_external: true,
      },
      tags: ['Tag1', 'Tag2'],
      theme: 'dark',
      attributes: 'data-test="true"',
      modifier_class: 'custom-class',
    });

    const element = c.querySelector('.ct-snippet');
    expect(element).not.toBeNull();
    expect(element.classList.contains('ct-theme-dark')).toBe(true);
    expect(element.classList.contains('custom-class')).toBe(true);
    expect(element.getAttribute('data-test')).toEqual('true');

    expect(c.querySelector('.ct-snippet__content-top').textContent.trim()).toEqual('Top content');
    expect(c.querySelector('.ct-snippet__content-bottom').textContent.trim()).toEqual('Bottom content');
    expect(c.querySelector('.ct-snippet__title').textContent.trim()).toContain('Snippet Title');
    expect(c.querySelector('.ct-snippet__summary').textContent.trim()).toEqual('This is the summary of the snippet.');
    expect(c.querySelector('.ct-snippet__content-middle').textContent.trim()).toEqual('Middle content');

    const link = c.querySelector('.ct-snippet__title__link');
    expect(link).toBeTruthy();
    expect(link.getAttribute('href')).toEqual('https://example.com/read-more');
    expect(link.getAttribute('target')).toEqual('_blank');

    const tags = c.querySelectorAll('.ct-snippet__tags .ct-tag');
    expect(tags).toHaveLength(2);
    expect(tags[0].textContent.trim()).toEqual('Tag1');
    expect(tags[1].textContent.trim()).toEqual('Tag2');

    assertUniqueCssClasses(c);
  });

  test('does not render when title is empty', async () => {
    const c = await dom(template, {
      title: '',
    });

    expect(c.querySelectorAll('.ct-snippet')).toHaveLength(0);
  });

  test('renders with link and tags', async () => {
    const c = await dom(template, {
      title: 'Snippet Title',
      link: {
        text: 'Read more',
        url: 'https://example.com/read-more',
      },
      tags: ['Tag1', 'Tag2'],
    });

    const link = c.querySelector('.ct-snippet__title__link');
    expect(link).toBeTruthy();
    expect(link.getAttribute('href')).toEqual('https://example.com/read-more');

    const tags = c.querySelectorAll('.ct-snippet__tags .ct-tag');
    expect(tags).toHaveLength(2);
    expect(tags[0].textContent.trim()).toEqual('Tag1');
    expect(tags[1].textContent.trim()).toEqual('Tag2');

    assertUniqueCssClasses(c);
  });

  test('renders with content slots', async () => {
    const c = await dom(template, {
      content_top: 'Top content',
      title: 'Snippet Title',
      content_middle: 'Middle content',
      content_bottom: 'Bottom content',
    });

    expect(c.querySelector('.ct-snippet__content-top').textContent.trim()).toEqual('Top content');
    expect(c.querySelector('.ct-snippet__content-bottom').textContent.trim()).toEqual('Bottom content');
    expect(c.querySelector('.ct-snippet__title').textContent.trim()).toEqual('Snippet Title');
    expect(c.querySelector('.ct-snippet__content-middle').textContent.trim()).toEqual('Middle content');

    assertUniqueCssClasses(c);
  });
});
