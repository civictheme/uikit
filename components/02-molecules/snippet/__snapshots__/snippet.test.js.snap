// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet Component does not render when title is empty 1`] = `
<div>
  


</div>
`;

exports[`Snippet Component renders with content slots 1`] = `
<div>
  

  
  <div
    class="ct-snippet ct-theme-light "
  >
    
          
    <div
      class="ct-snippet__content"
    >
      
                              
      <div
        class="ct-snippet__content-top"
      >
        Top content
      </div>
      
                  
                              
            


  
      <h4
        class="ct-heading ct-theme-light ct-snippet__title"
      >
                                      Snippet Title
                          
      </h4>
      
                  
                              
      <div
        class="ct-snippet__content-middle"
      >
        Middle content
      </div>
      
                  
                          
                          
                              
      <div
        class="ct-snippet__content-bottom"
      >
        Bottom content
      </div>
      
                        
    </div>
    
      
  </div>
  

</div>
`;

exports[`Snippet Component renders with link and tags 1`] = `
<div>
  

  
  <div
    class="ct-snippet ct-theme-light "
  >
    
          
    <div
      class="ct-snippet__content"
    >
      
                          
                              
            


  
      <h4
        class="ct-heading ct-theme-light ct-snippet__title"
      >
                                      


  
  
  
  
        <a
          class="ct-link ct-theme-light     ct-snippet__title__link"
          href="https://example.com/read-more"
        >
              Snippet Title  
        </a>
        
                          
      </h4>
      
                  
                          
                          
                              
      <div
        class="ct-snippet__tags"
      >
        
                              

  
        <div
          class="ct-tag-list ct-theme-light  "
        >
          
              
                                                                                          

  
          <ul
            class="ct-item-list ct-item-list--horizontal ct-item-list--small  ct-tag-list__content"
          >
            
                  
            <li
              class="ct-item-list__item"
            >
              
                      

   
  
  
    
              <span
                class="ct-tag ct-theme-light ct-tag--primary    "
              >
                     Tag1   
              </span>
              
          
        
            </li>
            
                        
            <li
              class="ct-item-list__item"
            >
              
                      

   
  
  
    
              <span
                class="ct-tag ct-theme-light ct-tag--primary    "
              >
                     Tag2   
              </span>
              
          
        
            </li>
            
            
          </ul>
          
          
                
        </div>
        
                          
      </div>
      
                  
                                
    </div>
    
      
  </div>
  

</div>
`;

exports[`Snippet Component renders with optional attributes 1`] = `
<div>
  

  
  <div
    class="ct-snippet ct-theme-dark custom-class"
    data-test="true"
  >
    
          
    <div
      class="ct-snippet__content"
    >
      
                              
      <div
        class="ct-snippet__content-top"
      >
        Top content
      </div>
      
                  
                              
            


  
      <h4
        class="ct-heading ct-theme-dark ct-snippet__title"
      >
                                      


  
        
  
  
        <a
          aria-label="Opens in a new tab"
          class="ct-link ct-theme-dark     ct-snippet__title__link"
          href="https://example.com/read-more"
          target="_blank"
        >
              Snippet Title        
          <span
            class="ct-visually-hidden"
          >
            (Opens in a new tab/window)
          </span>
          
    
        </a>
        
                          
      </h4>
      
                  
                              
      <div
        class="ct-snippet__content-middle"
      >
        Middle content
      </div>
      
                  
                              


      <div
        class="ct-paragraph ct-theme-dark ct-paragraph--regular ct-snippet__summary"
      >
        
    This is the summary of the snippet.  
      </div>
                        
                              
      <div
        class="ct-snippet__tags"
      >
        
                              

  
        <div
          class="ct-tag-list ct-theme-dark  "
        >
          
              
                                                                                          

  
          <ul
            class="ct-item-list ct-item-list--horizontal ct-item-list--small  ct-tag-list__content"
          >
            
                  
            <li
              class="ct-item-list__item"
            >
              
                      

   
  
  
    
              <span
                class="ct-tag ct-theme-dark ct-tag--primary    "
              >
                     Tag1   
              </span>
              
          
        
            </li>
            
                        
            <li
              class="ct-item-list__item"
            >
              
                      

   
  
  
    
              <span
                class="ct-tag ct-theme-dark ct-tag--primary    "
              >
                     Tag2   
              </span>
              
          
        
            </li>
            
            
          </ul>
          
          
                
        </div>
        
                          
      </div>
      
                  
                              
      <div
        class="ct-snippet__content-bottom"
      >
        Bottom content
      </div>
      
                        
    </div>
    
      
  </div>
  

</div>
`;

exports[`Snippet Component renders with required attributes 1`] = `
<div>
  

  
  <div
    class="ct-snippet ct-theme-light "
  >
    
          
    <div
      class="ct-snippet__content"
    >
      
                          
                              
            


  
      <h4
        class="ct-heading ct-theme-light ct-snippet__title"
      >
                                      Snippet Title
                          
      </h4>
      
                  
                          
                              


      <div
        class="ct-paragraph ct-theme-light ct-paragraph--regular ct-snippet__summary"
      >
        
    This is the summary of the snippet.  
      </div>
                        
                          
                                
    </div>
    
      
  </div>
  

</div>
`;
